components/SearchBar/index.js

```diff
@@ -0,0 +1,201 @@
+                onChange={(event) =>
+                  onSearch({ ...search, category: event.target.value })
+                }
+              >
+                <option value="">All</option>
+                <option value="Groceries">Groceries</option>
+                <option value="Housing">Housing</option>
+                <option value="Insurance">Insurance</option>
+              </StyledSearchInputSelect>
+            </StyledFilterItem>
+            <StyledFilterItem>
+              <label>Direction</label>
+              <StyledSearchInputSelect
+                value={search.direction}
+                onChange={(event) =>
+                  onSearch({ ...search, direction: event.target.value })
+                }
+              >
+                <option value="">All</option>
+                <option value="Income">Income</option>
+                <option value="Expense">Expense</option>
+              </StyledSearchInputSelect>
+            </StyledFilterItem>
+          </StyledFilterContainer>
+          <StyledFilterContainer>
+            <StyledFilterItem>
+              <label>Date Start</label>
+              <StyledSearchInput
+                type="date"
+                value={search.dateFrom}
+                onChange={(event) =>
+                  onSearch({ ...search, dateFrom: event.target.value })
+                }
+              />
+            </StyledFilterItem>
+            <StyledFilterItem>
+              <label>Date End</label>
+              <StyledSearchInput
+                type="date"
+                value={search.dateTo}
+                onChange={(event) =>
+                  onSearch({ ...search, dateTo: event.target.value })
+                }
+              />
+            </StyledFilterItem>
+          </StyledFilterContainer>
+          <StyledFilterContainer>
+            <StyledFilterItem>
+              <label>Amount From</label>
+              <StyledSearchInput
+                type="number"
+                placeholder="Search by Amount From"
+                value={search.amountFrom}
+                onChange={(event) =>
+                  onSearch({ ...search, amountFrom: event.target.value })
+                }
+              />
+            </StyledFilterItem>
+            <StyledFilterItem>
+              <label>Amount To</label>
+              <StyledSearchInput
+                type="number"
+                placeholder="Search by Amount To"
+                value={search.amountTo}
+                onChange={(event) =>
+                  onSearch({ ...search, amountTo: event.target.value })
+                }
+              />
+            </StyledFilterItem>
+            <StyledFilterItem>
+              <label>X</label>
+              <StyledFilterButton onClick={handleResetFilters}>
+                Clear Filter
+              </StyledFilterButton>
+            </StyledFilterItem>
+          </StyledFilterContainer>
+        </>
+      ) : (
+        <></>
+      )}
```

Comment:
since you don't display anything if `isFilter` is false you can use `&&` instead of the ternary operator: 

```js
isFilter && <> ... </>
```

Reply:
thanks for the tip, that's right that's a better way of writing